### PR TITLE
P1522R1 Iterator difference type and integer overflow

### LINKED ISSUE #3029: [2019-07 LWG Motion 25] P1522R1 Iterator Difference Type and Integer Overflow

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #3029.

### CODE REVIEW

File: source/ranges.tex [hunk 1]
```diff
@@ -1752,39 +1780,39 @@
...
       { i += n } -> Same<I&>;
       { i -= n } -> Same<I&>;
-      { j +  n } -> Same<I>;
-      { n +  j } -> Same<I>;
-      { j -  n } -> Same<I>;
-      { j -  j } -> Same<iter_difference_t<I>>;
+      I(j + n);
+      I(n + j);
+      I(j - n);
+      { j -  j } -> ConvertibleTo<@\placeholdernc{IOTA_DIFF_T}@(I)>;
```
Comment (johelegp) [inline source/ranges.tex]: The extra space for alignment before the second operand could be removed, or added to the above expressions.

File: source/ranges.tex [hunk 2]
```diff
@@ -1752,39 +1780,39 @@
...
     };
 \end{itemdecl}
 
+Let \tcode{D} be \tcode{\placeholdernc{IUOTA_DIFF_T}(I)}.
```
Comment (johelegp) [inline source/ranges.tex]: IOTA, not IUOTA.

File: source/ranges.tex (hunk 1 above)
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

File: source/ranges.tex (hunk 2 above)
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

**Pull request merged**